### PR TITLE
NC-14

### PR DESCRIPTION
**Pull request by BillyZen**
/api/articles/:article_id endpoint and error handling

### CODE REVIEW

File: Controllers/controllers.js [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
         next(err)
     })
+}
+
+exports.getArticle = (req, res, next) => {
+    const id = req.params.article_id
+    selectArticle(id)
+    .then(article => {
+        console.log(article)
+        res.status(200).send({article})
```
Comment (daviidbartlett) [inline Controllers/controllers.js:17]: remove console.log

File: Models/models.js [hunk 2]
```diff
@@ -5,4 +5,11 @@ exports.selectTopics = () => {
     .then(({rows}) => {
         return rows
     })
+}
+
+exports.selectArticle = (id) => {
+    return db.query('SELECT * FROM articles WHERE article_id = $1;', [id])
+    .then(response => {
+        return response.rows
+    })
```
Comment (daviidbartlett) [inline Models/models.js:21]: I wouldn't expect this as an array

File: __tests__/app.test.js [hunk 3]
```diff
@@ -36,4 +35,44 @@ describe("/api/topics", () => {
...
+
+describe("/api/articles/:article_id", () => {
+	describe("GET", () => {
+        test(`status: 200 returns an article object, which should have the following properties:
+            authors (which is the username for the users table), title, article_id, body, topic, created_at, votes`, () => {
+            return request(app)
+            .get("/api/articles/1")
+            .expect(200)
+            .then(response => {
+                expect(response.body.article).toHaveLength(1)
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: no array please

File: __tests__/app.test.js [hunk 4]
```diff
@@ -36,4 +35,44 @@ describe("/api/topics", () => {
...
+						}])
+            });
+        });
+        test("status: 404 returns an error message of 404: path not found", () => {
+		return request(app)
+            .get("/api/article")
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe('Path not found')
+	        });
```
Comment (daviidbartlett) [inline __tests__/app.test.js]: I expect a 404 but not for a path not found

**Review by daviidbartlett — COMMENTED** [review]
missing a test and a few comments:
- missing article not found

File: Controllers/controllers.js (hunk 1 above)
Comment (BillyZen) [reply Controllers/controllers.js:17]: Done

File: Models/models.js (hunk 2 above)
Comment (BillyZen) [reply Models/models.js:21]: Sorted

File: __tests__/app.test.js (hunk 3 above)
Comment (BillyZen) [reply __tests__/app.test.js]: Sorted

File: __tests__/app.test.js (hunk 4 above)
Comment (BillyZen) [reply __tests__/app.test.js]: Created custom errors controller function and all tests passing

File: Controllers/errors.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  if (err.code === '22P02') {
+    res.status(400).send({ msg: 'Bad Request' });
+  } else next(err);
+}
+
+
+exports.handle500 = (err, req, res, next) => {
+    console.log(err)
+    res.status(500).send({ msg: "Internal Server Error" });
+};
```
Comment (Angaca) [inline Controllers/errors.js:17]: File nicely organised!

**Review by Angaca — APPROVED** [review]
Looks good now!

**Pull request merged**